client/src/components/Attachment/UploadAttachment.js

```diff
@@ -36,6 +35,25 @@ const UploadAttachment = ({
   const [remove, setRemove] = useState(false)
   const [uploadedList, setUploadedList] = useState([])
 
+  useEffect(() => {
+    // In the edit mode the base page entity (report, location, etc.) always has an uuid
+    // So when uploading new attachment it is always inserted related tables, (ie. see attachmentSave)
+    if (edit) {
+      return
+    }
+    // if base entity is not created, the attachment can not be associated
+    if (!relatedObjectUuid) {
+      return
+    }
+    // Update only object not associated with the base entity, as in report
+    const updateList = uploadedList.filter(
+      list => !list.attachmentRelatedObjects[0].relatedObjectUuid
```

Comment:
It is unclear to me what this effect is supposed to accomplish. And this specific statement implies that attachments can have only one related object, which is not the general case.